### PR TITLE
Add agent mTLS auth explanation for secrets docs

### LINKED ISSUE #2226: Clarify the requirement of mTLS authentication when using sensu go secrets on agents
### Description
When setting up Sensu secrets, language in at least the secrets reference may lead users to believe that only TLS is required for it to function:

https://docs.sensu.io/sensu-go/latest/reference/secrets/

```
Secrets are only transmitted over a TLS websocket connection. Unencrypted connections must not transmit privileged information. For agent-side resources, enable TLS/mTLS.
```

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [X] improvement (improve docs content or format)

### Affected pages
- At least the secrets reference.
- Any guide that mentions on how to setup secrets with agent examples.
- Maybe categorizing pre-5.14 TLS and mTLS.

### Expected behavior
Need more clear language that mTLS (mutual TLS) is required for agent resources. It is called out briefly on the reference page `For agent-side resources, enable TLS/mTLS`, but we should have links to exactly what it means to enable mTLS and that it does not include the TLS configuration that pre-dated 5.14.

### Current behavior
Before mTLS was added in 5.14, we have other ways to encrypt traffic between the agent and backend using wss. This can lead to confusion on exactly what is required for agent secrets to be transmitted by the backend to the agent requesting them.

### Corrections needed
Needs further discussion on each area that needs this due to previous use cases of wss prior to 5/14.

### Additional context

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds explanation and terminology updates to the following docs:
- https://docs.sensu.io/sensu-go/latest/guides/securing-sensu/#sensu-agent-tls-authentication
- https://docs.sensu.io/sensu-go/latest/guides/secrets-management/
- https://docs.sensu.io/sensu-go/latest/reference/secrets/
- https://docs.sensu.io/sensu-go/latest/reference/secrets-providers/

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2226

## Review Instructions
I think this draft PR may only scratch the surface of the changes required to clarify mTLS and related secrets management requirements.

If I understand the issue correctly, we also need to add to the content in https://docs.sensu.io/sensu-go/latest/guides/securing-sensu/#sensu-agent-tls-authentication so that it's clearer how users should enable mTLS authentication for agents. There was also a change at 5.14 related to TLS vs. mTLS that we may need to explain as well.

I am not sure if further changes are needed for the secrets management guide, secrets reference, and secrets providers reference.

@rgeniesse What do you think? Maybe it would be best to set up a call to go over this?

### CODE REVIEW

**Comment by hillaryfraley** [comment]
I found my notes from our retro meeting last week when Richie and Cameron discussed this:

- It's not enough to have agents communicate over TLS to use secrets.
- Agent doesn't connect and send username and password; agent connects and sends its own certificates.
- This ensures the agent receiving the secret is authorized to receive it.
- Expand to generate certificates for mTLS (Generate Certificates guide in progress: https://github.com/sensu/sensu-docs/pull/2212)

File: content/sensu-go/5.18/guides/secrets-management.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ Secrets are configured via [secrets resources][8].
 A secret resource definition refers to the secrets provider (`Env` or `VaultProvider`) and an ID (the named secret to fetch from the secrets provider).
 
 This guide only covers the handler use case, but you can use secrets management in handler, mutator, and check execution.
-For secrets management in checks, the Sensu backend will transmit requests over its secure transport (TLS-encrypted websockets) to your Sensu agent to execute your check, so mTLS must be enabled.
+For secrets management in checks, the Sensu backend will transmit requests over its secure transport [transport layer security (TLS)-encrypted websockets] to your Sensu agent to execute your check, so you must [enable mutual TLS (mTLS)][15].
```
Comment (cwjohnston) [inline content/sensu-go/5.18/guides/secrets-management.md]:
```suggestion
When secrets are referenced in check configurations, the Sensu backend will only transmit execution requests for those checks to agents connected via [mutually-authenticated transport layer security (mTLS)-encrypted websockets]. Read more about [enabling mutual TLS (mTLS)][15].
```

It’s not enough for agents to be connected via TLS, their TLS session must be mutually authenticated.

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.18/guides/secrets-management.md [hunk 2]
```diff
@@ -37,7 +37,7 @@ Secrets are configured via [secrets resources][8].
 A secret resource definition refers to the secrets provider (`Env` or `VaultProvider`) and an ID (the named secret to fetch from the secrets provider).
 
 This guide only covers the handler use case, but you can use secrets management in handler, mutator, and check execution.
-For secrets management in checks, the Sensu backend will transmit requests over its secure transport (TLS-encrypted websockets) to your Sensu agent to execute your check, so mTLS must be enabled.
+When secrets are referenced in check configurations, the Sensu backend will only transmit execution requests for those checks to agents connected via [mutually-authenticated transport layer security (mTLS)-encrypted websockets]. Read more about [enabling mutual TLS (mTLS)][15].
```
Comment (hillaryfraley) [inline content/sensu-go/5.18/guides/secrets-management.md]:
```suggestion
When a check configuration references a secret, the Sensu backend will only transmit the check's execution requests to agents that are connected via mutually authenticated transport layer security (mTLS)-encrypted websockets. Read more about [enabling mTLS][15].
```

**Comment by hillaryfraley** [comment]
Seems like I should add the mTLS info here too: https://docs.sensu.io/sensu-go/latest/reference/agent/#communication-between-the-agent-and-backend

**Comment by hillaryfraley** [comment]
I moved this one out of draft status because I think we're at the content review stage.

File: content/sensu-go/5.18/guides/securing-sensu.md [hunk 3]
```diff
@@ -177,7 +178,7 @@ Certificate:
 
 The `Subject:` field indicates the certificate's CN is `client`, so to bind the agent to a particular user in Sensu, create a user called `client`.
 
-To enable agent TLS authentication, use existing certificates and keys for the Sensu backend and agent or create new certificates and keys according to the [Create self-signed certificates][12] section.
+To enable agent mTLS authentication, use existing certificates and keys for the Sensu backend and agent or create new certificates and keys according to the [Create self-signed certificates][12] section.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/guides/securing-sensu.md]:
```suggestion
To enable agent mTLS authentication, create and distribute new certificates and keys according to the [Create self-signed certificates][12] section. Once the TLS certificate and key are in place, [update the agent configuration using `cert-file` and `key-file` security configuration flags][7].
```

File: content/sensu-go/5.18/guides/securing-sensu.md [hunk 4]
```diff
@@ -277,6 +278,8 @@ client.pem
 
 Learn about [role-based access control (RBAC) in Sensu][3] or [create a read-only user][4].
 
+
+[1]: ../../guides/secrets-management/
```
Comment (cwjohnston) [inline content/sensu-go/5.18/guides/securing-sensu.md]:
```suggestion
[1]: ../../guides/secrets-management/
[7]: ../../reference/agent/#security-configuration-flags
```

Comment (cwjohnston) [reply content/sensu-go/5.18/guides/securing-sensu.md]: Also need to update #12 to ../../guides/generate-certificates/

File: content/sensu-go/5.18/reference/secrets.md [hunk 5]
```diff
@@ -29,9 +29,11 @@ Only Sensu backends have access to request secrets from a [secrets provider][7].
 Sensu backends cache fetched secrets in memory, with no persistence to a Sensu datastore or file on disk.
 Secrets provided via a "lease" with a "lease duration" are deleted from Sensu's in-memory cache after the configured number of seconds, prompting the Sensu backend to request the secret again.
 
-Secrets are only transmitted over a TLS websocket connection.
+Secrets are only transmitted over a transport layer security (TLS) websocket connection.
 Unencrypted connections must not transmit privileged information.
-For agent-side resources, enable TLS/mTLS.
+
+For agent-side resources, you must [enable mutual TLS (mTLS)][13].
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/secrets.md]: I don't think "agent-side resources" is clear. Which resources are these? I think it means checks, hooks and assets.

File: content/sensu-go/5.18/guides/securing-sensu.md [hunk 6]
```diff
@@ -49,7 +49,7 @@ The Sensu Go Agent API, HTTP API, and web UI use a common stanza in `/etc/sensu/
 Here are the attributes you'll need to configure.
 
 {{% notice note %}}
-**NOTE**: By changing these parameters, the server will communicate over TLS and expect agents that connect to it to use the WebSocket secure protocol.
+**NOTE**: By changing these parameters, the server will communicate over transport layer security (TLS) and expect agents that connect to it to use the WebSocket secure protocol.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/guides/securing-sensu.md]:
```suggestion
**NOTE**: By changing these parameters, the server will communicate using transport layer security (TLS) and expect agents that connect to it to use the WebSocket secure protocol.
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.18/guides/securing-sensu.md [hunk 7]
```diff
@@ -177,7 +178,7 @@ Certificate:
 
 The `Subject:` field indicates the certificate's CN is `client`, so to bind the agent to a particular user in Sensu, create a user called `client`.
 
-To enable agent TLS authentication, use existing certificates and keys for the Sensu backend and agent or create new certificates and keys according to the [Create self-signed certificates][12] section.
+To enable agent mTLS authentication, create and distribute new certificates and keys according to the [Create self-signed certificates][12] section. Once the TLS certificate and key are in place, [update the agent configuration using `cert-file` and `key-file` security configuration flags][7].
```
Comment (hillaryfraley) [inline content/sensu-go/5.18/guides/securing-sensu.md]:
```suggestion
To enable agent mTLS authentication, create and distribute new certificates and keys according to the [Generate certificates][12] guide.
Once the TLS certificate and key are in place, [update the agent configuration using `cert-file` and `key-file` security configuration flags][7].
```

File: content/sensu-go/5.18/reference/secrets.md (hunk 5 above)
Comment (hillaryfraley) [inline content/sensu-go/5.18/reference/secrets.md]:
```suggestion
For checks, hooks, and assets, you must [enable mutual TLS (mTLS)][13].
```

**Review by cwjohnston — APPROVED** [review]

**Pull request merged**